include/mppic/critics/obstacles_critic.hpp

```diff
@@ -22,20 +20,29 @@ class ObstaclesCritic : public CriticFunction
    * @param costs [out] add obstacle cost values to this tensor
    */
   virtual void score(
-    const geometry_msgs::msg::PoseStamped & /*robot_pose*/, const xt::xtensor<double, 3> & trajectories,
-    const xt::xtensor<double, 2> & /*path*/, xt::xtensor<double, 1> & costs) override;
+    const geometry_msgs::msg::PoseStamped & /*robot_pose*/,
+    const xt::xtensor<double, 3> & trajectories, const xt::xtensor<double, 2> & /*path*/,
+    xt::xtensor<double, 1> & costs) override;
 
 protected:
-  unsigned char costAtPose(const auto & point);
-
-  bool inCollision(unsigned char & cost) const;
-
-  bool isFree(unsigned char & cost) const;
-
-  double toDist(unsigned char & cost);
-
-  double scoreDistance(double & min_dist);
+  bool inCollision(unsigned char cost) const;
+  double toDist(unsigned char cost);
+  double scoreCost(unsigned char cost);
+  unsigned char maxCost();
+
+  unsigned char costAtPose(const auto & point)
+  {
+    unsigned char cost;
```

Comment:
In cpp file would be great, else I'll do it in my next PR

Reply:
why? I don't see anything in the function requiring being in the hpp file, that's all pretty standard. If its `auto` for the `point` type, this is only used in 1 place so that could be explicitly typed

Reply:
it has type something like 
"xt::xfunction<xt::detail::plus, xt::xview<xt::xarray_container<xt::uvector<double, std::allocator<double> >, (xt::layout_type)1, xt::svector<unsigned long, 4ul, std::allocator<unsigned long>, true>, xt::xtensor_expression_tag>&, int>, xt::xarray_container<xt::uvector<double, std::allocator<double> >, (xt::layout_type)1, xt::svector<unsigned long, 4ul, std::allocator<unsigned long>, true>, xt::xtensor_expression_tag> const&>"

we could try to decltype it however

Reply:
Oh. Uh. Ok yeah we can leave as is. Why is it that this is a problem now but not before?

Reply:
But i dont have quick solution here. I think it's simpler to just let it as it is or make it lambda